src/Streamly/Internal/Data/Parser/ParserD/Type.hs

```diff
@@ -424,7 +424,7 @@ noErrorUnsafeSplitWith func (Parser stepL initialL extractL)
             -- Assume that the first parser can never fail, therefore we do not
             -- need to keep the input for backtracking.
             Partial n s -> return $ Partial n (SeqParseL s)
-            Continue n s -> return $ Partial n (SeqParseL s)
+            Continue n s -> return $ Continue n (SeqParseL s)
```

Comment:
This should be `Partial`. `noErrorUnsafeSplitWith` isf or parsers that don't fail.

Reply:
Continue does not mean failure, its only backtracking. The inner parser wants to backtrack but you are dropping the backtrack buffer and continuing from the current position that would make the behavior of the parser incorrect.